browser/IgBlazorSamples.Gulp/tasks/gulp-samples.js

```diff
@@ -809,18 +808,60 @@ function updateCodeViewer(cb) {
     for (const sample of samples) {
-        log('updateCodeViewer for ' + sample.SourceFiles[0].Path);
 
-       // - generate .json file with info about source files, e.g.
-       //     \browser\IgBlazorSamples.Client\wwwroot\code-viewer\charts-category-chart-annotations.json
-       // - note that "content" field should contain source code for sample files: .razor/.tsx, .cs/.ts, and .css
-       // - save all .json files in this folder: \browser\IgBlazorSamples.Client\wwwroot\code-viewer\
-       // - save each .json file with sample's component name and sample's folder name: charts-category-chart-annotations.json
+        var content = "{ \r\n \"sampleFiles\": [ \r\n";
 
+        for (const file of sample.SourceFiles) {
+            if (file.isRazorSample()) {
+                content += constructCodeViewerItem(file.Path, file.Content);
+            }
+            else if (file.isCS()) {                
+                content += constructCodeViewerItem(file.Path, file.Content);
+            }
+        }
+    
+        if (sample.PublicFiles_JS.length > 0) {
+            for(const file of sample.PublicFiles_JS){                
+                content += constructCodeViewerItem(file.Path, file.Content);
+            }
+        }
+    
+        if (sample.PublicFiles_CSS.length > 0) {
+            for(const file of sample.PublicFiles_CSS){                
+                content += constructCodeViewerItem(file.Path, file.Content);
+            }    
+        }
+    
+        content += "] \r\n }";
+    
+        var path = sample.SampleRoute;
+        path = path.substring(1, path.length);
+        path = path.replace("/", "-");
+    
+        var codeViewPath = "../IgBlazorSamples.Client/wwwroot/code-viewer/" + path + ".json";
+    
+        fs.writeFileSync(codeViewPath, content);
+
+        // - generate .json file with info about source files, e.g.
+        //     \browser\IgBlazorSamples.Client\wwwroot\code-viewer\charts-category-chart-annotations.json
+        // - note that "content" field should contain source code for sample files: .razor/.tsx, .cs/.ts, and .css
+        // - save all .json files in this folder: \browser\IgBlazorSamples.Client\wwwroot\code-viewer\
+        // - save each .json file with sample's component name and sample's folder name: charts-category-chart-annotations.json
     }
 
-    // run 'gulp updateCodeViewer' to execute this function
-
     cb();
 
-} exports.updateCodeViewer = updateCodeViewer;
\ No newline at end of file
+} exports.updateCodeViewer = updateCodeViewer;
+
+function constructCodeViewerItem(filePath, content) {
+
+    var JSONString = JSON.stringify(content);
+
+    var actualContent = JSONString.replace("\r\n/g", "\\r\\n");
+
+    var relativeAssetsUrls = "{ \r\n \"hasRelativeAssetsUrls\": false, \r\n";
+    var path = "\"path\": \"" + filePath + "\",\r\n";
+    var content = "\"content\": " + actualContent + "\r\n }, \r\n";
```

Comment:
your JSON files are missing this kind of entries for each file:
```      
"isMain" : true,
"fileExtension" : "ts",
"fileHeader" : "ts"
 ```
where isMain should be true if it is sample file (.razor, .css, .html, .js) and false when it is data file (.cs)